runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java

```diff
@@ -234,6 +237,26 @@ public void getState(
     }
   }
 
+  @Override
+  public void getPipeline(
+      GetJobPipelineRequest request, StreamObserver<GetJobPipelineResponse> responseObserver) {
+    LOG.trace("{} {}", GetJobPipelineRequest.class.getSimpleName(), request);
+    String invocationId = request.getJobId();
+    try {
+      JobInvocation invocation = getInvocation(invocationId);
+      RunnerApi.Pipeline pipeline = invocation.getPipeline();
+      GetJobPipelineResponse response =
+          GetJobPipelineResponse.newBuilder().setPipeline(pipeline).build();
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      String errMessage =
+          String.format("Encountered Unexpected Exception for Invocation %s", invocationId);
+      LOG.error(errMessage, e);
```

Comment:
nit: SLF4J allows you to do
```
LOG.error(Encountered unexpected exception for invocation {}", invocationId, e); 
```
instead of using `String.format`

Reply:
Again, this pattern is repeated in 5 other places in this file.  I was just trying to stay consistent.  

If you don't like the style used in the file, I would recommend addressing them all at once in another PR.

Reply:
Sounds good to me.